### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by karthikdilliraj**
Hi Guys, this is the new PR. We have made many changes, please do review them ASAP.

### CODE REVIEW

File: include/calculate_fusion.h [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ */
+struct eigen_systems
+{
+    double *eigen_value;
+    double **eigen_vector;
+};
+
+/**
+ * @brief Calculate Support degree matrix D for the set of sensor values
+ *
```
Comment (ZiyadR) [inline include/calculate_fusion.h]: You need to include a detailed description in addition to the brief description for each function.

File: src/sensor.c [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+        return INVALID_SENSOR_FUSION_VALUE;
+    }
+
+    /*
+    * Step 7 - eliminate_incorrect_data
+    */
+    int result_eliminate = eliminate_incorrect_data(integrated_support_matrix,
+                                                    ((float)q_support_value / 100.0), no_of_sensors);
+
+    if (result_eliminate < 0)
```
Comment (ZiyadR) [inline src/sensor.c]: You should start the curly bracket at the end of the first line.

Comment (karthikdilliraj) [reply src/sensor.c]: Changes have been done

**Comment by karthikdilliraj** [comment]
> I tested the program using multiple input files. The fused value for the last timestamp in the file is always wrong and extremely large.

File: include/calculate_fusion.h (hunk 1 above)
Comment (karthikdilliraj) [reply include/calculate_fusion.h]: Detailed descriptions are added

**Comment by karthikdilliraj** [comment]
> I tested the program using multiple input files. The fused value for the last timestamp in the file is always wrong and extremely large.

Do check the docs.md file, for the usage of running the executable file

**Comment by ZiyadR** [comment]
> > I tested the program using multiple input files. The fused value for the last timestamp in the file is always wrong and extremely large.
> 
> Do check the docs.md file, for the usage of running the executable file

I ran it using the default values and using the sample input.csv file in the src folder.
The last fused value was clearly incorrect.

**Comment by HenryLeCarleton** [comment]
> > > I tested the program using multiple input files. The fused value for the last timestamp in the file is always wrong and extremely large.
> > 
> > 
> > Do check the docs.md file, for the usage of running the executable file
> 
> I ran it using the default values and using the sample input.csv file in the src folder.
> The last fused value was clearly incorrect.

@ZiyadR  May I ask what the correct value to you

**Pull request merged**